vllm/v1/engine/detokenizer.py

```diff
@@ -156,21 +196,20 @@ def add_tokens(
         delta = self.output_kind == RequestOutputKind.DELTA
         output_text = self._get_next_output_text(finished, delta)
         token_ids = new_token_ids if delta else self.output_token_ids
+        logprobs = new_logprobs if delta else self.logprobs
+        prompt_logprobs = new_prompt_logprobs if delta else self.prompt_logprobs
 
         request_output = RequestOutput.new(
             self.request_id,
             self.prompt,
             self.prompt_token_ids,
             output_text,
             token_ids,
+            logprobs,
+            prompt_logprobs,
             finished,
         )
 
-        if finished:
-            completion_output = request_output.outputs[0]
```

Comment:
Why do we remove the `finish` and `stop_reason`?

Reply:
Woops. Initially I puzzled over this and thought completion output was not being used & this was just some junk code left over from a previous iteration of the design. Now I realized the completion output within the request output is being accessed & edited in-place. So, I went and added these lines back. Thanks!